rasa/nlu/extractors/lookup_entity_extractor.py

```diff
@@ -0,0 +1,119 @@
+
+import rasa.utils.common as common_utils
+from rasa.nlu.constants import ENTITIES
+from rasa.nlu.training_data import Message
+from rasa.nlu.extractors.extractor import EntityExtractor
+
+logger = logging.getLogger(__name__)
+
+
+class LookupEntityExtractor(EntityExtractor):
+    """
+    Searches for entities in the user's message from a list of examples.
+    Required Parameters:
+    @lookup -> dict
+    """
+
+    defaults = {
+        # lookup key for extracting lookup entities,
+        # it contains the dictonary of lookup entity names
+        # and their respective data files
+        # example:
+        # - name: LookupEntityExtractor
+        #   lookup:
+        #      city: /some/path/city.txt
+        #      person: /some/other/path/person.txt
+        "lookup": None
+    }
+
+    def __init__(self, component_config: Optional[Dict[Text, Any]] = None):
+        super(LookupEntityExtractor, self).__init__(component_config)
+
+        if component_config is not None and "lookup" in component_config:
+            if component_config["lookup"] is not None:
+                # check if the entities and respective file path exists
+                for key, value in list(component_config["lookup"].items()):
+                    self.add_entities_if_lookup_entity(key, value)
+            else:
+                self.component_config["lookup"] = None
+
+            if not bool(component_config["lookup"]):
+                # if the lookup dictionary is empty after filtering
+                # the lookup entities, assign it None value
+                self.component_config["lookup"] = None
+        else:
+            common_utils.raise_warning(
+                "Can't extract Lookup Entities,\
+                Please configure the lookup entities in the config.yml."
+            )
+
+    def add_entities_if_lookup_entity(self, entity, file_path) -> None:
+        if file_path is not None:
+            if os.path.isfile(file_path):
+                pass
+            else:
+                # remove the entity from the lookup dictionary,
+                # if the file path doesn't exist
+                self.component_config["lookup"].pop(entity)
+                common_utils.raise_warning(
+                    f"can't extract lookup entity: '{entity}',\
+                    make sure the provided file '{file_path}' exists.")
+        else:
+            # remove the entity from the lookup dictionary,
+            # if the file path is not provided
+            self.component_config["lookup"].pop(entity)
+            common_utils.raise_warning(
+                f"can't extract '{entity}' entity,\
+                please provide the example file.")
+
+    def _parse_entities(self, user_input: Text) -> List[Dict[Text, Any]]:
+        """Extract entities from the user input."""
+        if self.component_config["lookup"] is not None:
+            for entity, file_path in list(
+                    self.component_config["lookup"].items()):
```

Comment:
```suggestion
            for entity, file_path in self.component_config["lookup"].items():
```
I don't think you need to wrap this in a list.